File: Assets/Scripts/Progression/EquipmentSlot.cs
using UnityEngine;

namespace ActionGameDemo.Progression
{
    public class EquipmentSlot
    {
        private EquipmentData _equippedItem;
        private readonly EquipmentSlotType _slotType;

        public EquipmentSlotType SlotType => _slotType;
        public EquipmentData EquippedItem => _equippedItem;
        public bool HasItem => _equippedItem != null;

        public EquipmentSlot(EquipmentSlotType slotType)
        {
            _slotType = slotType;
        }

        public EquipmentData Equip(EquipmentData item)
        {
            if (item == null || item.slotType != _slotType) return null;
            EquipmentData previous = _equippedItem;
            _equippedItem = item;
            return previous;
        }

        public EquipmentData Unequip()
        {
            EquipmentData removed = _equippedItem;
            _equippedItem = null;
            return removed;
        }
    }
}

File: server/src/main/java/com/actiongame/server/domain/progression/EquipmentSlot.java
package com.actiongame.server.domain.progression;

import com.actiongame.server.config.EquipmentConfig;

/**
 * 装备槽 (服务端权威)
 * 对应Unity EquipmentSlot
 */
public class EquipmentSlot {
    private final EquipmentSlotType slotType;
    private EquipmentConfig equippedItem;

    public EquipmentSlot(EquipmentSlotType slotType) {
        this.slotType = slotType;
    }

    public EquipmentConfig equip(EquipmentConfig item) {
        if (item == null || item.getSlotType() != slotType) return null;
        EquipmentConfig previous = equippedItem;
        equippedItem = item;
        return previous;
    }

    public EquipmentConfig unequip() {
        EquipmentConfig removed = equippedItem;
        equippedItem = null;
        return removed;
    }

    public EquipmentSlotType getSlotType() { return slotType; }
    public EquipmentConfig getEquippedItem() { return equippedItem; }
    public boolean hasItem() { return equippedItem != null; }
}
